torch/_dynamo/variables/tensor.py

```diff
@@ -1447,32 +1450,46 @@ def call_function(
         args: list[VariableTracker],
         kwargs: dict[str, VariableTracker],
     ) -> VariableTracker:
-        # Handle `Subclass(existing_tensor)` calls.
-        def impl():
-            if len(args) == 1 and isinstance(args[0], TensorVariable):
-                from .torch_function import TensorWithTFOverrideVariable
-
-                # This simulates `__new__` and _assumes_ it doesn't have
-                # side-effects that matters to Dynamo tracing. TODO trace through
-                # `__new__`.
+        # Handle `Subclass(existing_tensor, ...)` calls.
+        from .torch_function import TensorWithTFOverrideVariable
+
+        new_func = self.value.__new__
+        if new_func is torch.Tensor.__new__:
+            if (
+                len(args) == 1
+                and isinstance(args[0], TensorVariable)
+                and len(kwargs) == 0
+            ):
+                data = args[0]
+                # Simulate `torch.Tensor.__new__` as shallow-copying the input
+                # tensor data with a new type. TODO polyfill?
                 var = TensorWithTFOverrideVariable.from_tensor_var(
-                    tx, args[0], self.value, self.source
+                    tx, data, self.value, self.source
                 )
+            else:
+                unimplemented_v2(
+                    gb_type="Calling subclass default constructor with more than tensor argument",
+                    context=f"{self.value}(args={args}, kwargs={kwargs})",
+                    explanation="Currently not supported",
+                    hints=[
+                        "Avoid this constructor call or move it outside "
+                        "`torch.compile` regione",
+                        *graph_break_hints.SUPPORTABLE,
+                    ],
+                )
+        else:
+            # Let Dynamo trace through custom `__new__`
+            var = VariableTracker.build(tx, new_func).call_function(
+                tx, [self] + args, kwargs
+            )
```

Comment:
A torch-function-style tensor subclass custom `__new__` (as show in tests) typically consists of some `_make_subclass` call to construct the instance, and optional attribute mutations. This PR handles the `_make_subclass` part, and #149482 handles the attribute mutations part, thus we can safely trace through the custom `__new__`.